From ac14ffc4c2631d96b5fa82af4e39d31d9eecc741 Mon Sep 17 00:00:00 2001
From: Ozkan Sezer <sezeroz@gmail.com>
Date: Sun, 1 Dec 2024 04:33:50 +0300
Subject: [PATCH] update configure.ac aarch64, arm64, and amd64 to match as FPM_64BIT

---
 config.h.in  |   25 +++++++++++++++++++------
 configure.ac |    3 ++-
 2 files changed, 21 insertions(+), 7 deletions(-)

diff --git a/configure.ac b/configure.ac
index 3261fed..5cb3229 100644
--- a/configure.ac
+++ b/configure.ac
@@ -309,7 +309,8 @@ if test -z "$FPM" && test "$GCC" = yes
 then
     case "$host" in
 	i?86-*)     FPM="INTEL"  ;;
-	x86_64*)    FPM="64BIT"  ;;
+	amd64*|x86_64*)  FPM="64BIT"  ;;
+	arm64*|aarch64*) FPM="64BIT"  ;;
 	arm*-*)     FPM="ARM"    ;;
 	mips*-*)    FPM="MIPS"   ;;
 	sparc*-*)   FPM="SPARC"  ;;
diff --git a/config.h.in b/config.h.in
index 288b49d..e40f893 100644
--- a/config.h.in
+++ b/config.h.in
@@ -1,5 +1,8 @@
 /* config.h.in.  Generated from configure.ac by autoheader.  */
 
+/* Define if building universal (internal helper macro) */
+#undef AC_APPLE_UNIVERSAL_BUILD
+
 /* Define to enable diagnostic debugging support. */
 #undef DEBUG
 
@@ -69,8 +72,7 @@
 /* Define to 1 if you have the `waitpid' function. */
 #undef HAVE_WAITPID
 
-/* Define to the sub-directory in which libtool stores uninstalled libraries.
-   */
+/* Define to the sub-directory where libtool stores uninstalled libraries. */
 #undef LT_OBJDIR
 
 /* Define to disable debugging assertions. */
@@ -104,6 +106,9 @@
 /* Define to the one symbol short name of this package. */
 #undef PACKAGE_TARNAME
 
+/* Define to the home page for this package. */
+#undef PACKAGE_URL
+
 /* Define to the version of this package. */
 #undef PACKAGE_VERSION
 
@@ -113,9 +118,17 @@
 /* Version number of package */
 #undef VERSION
 
-/* Define to 1 if your processor stores words with the most significant byte
-   first (like Motorola and SPARC, unlike Intel and VAX). */
-#undef WORDS_BIGENDIAN
+/* Define WORDS_BIGENDIAN to 1 if your processor stores words with the most
+   significant byte first (like Motorola and SPARC, unlike Intel). */
+#if defined AC_APPLE_UNIVERSAL_BUILD
+# if defined __BIG_ENDIAN__
+#  define WORDS_BIGENDIAN 1
+# endif
+#else
+# ifndef WORDS_BIGENDIAN
+#  undef WORDS_BIGENDIAN
+# endif
+#endif
 
 /* Define to empty if `const' does not conform to ANSI C. */
 #undef const
@@ -129,5 +142,5 @@
 /* Define to `int' if <sys/types.h> does not define. */
 #undef pid_t
 
-/* Define to `unsigned' if <sys/types.h> does not define. */
+/* Define to `unsigned int' if <sys/types.h> does not define. */
 #undef size_t
